src/views/Range/hooks.tsx

```diff
@@ -0,0 +1,412 @@
+import { useTestableNetworks } from "src/hooks/useTestableNetworks";
+import { error as createErrorToast, info as createInfoToast } from "src/slices/MessagesSlice";
+import { BondTeller__factory, IERC20__factory } from "src/typechain";
+import { OlympusRange } from "src/typechain/Range";
+import { useNetwork, useSigner } from "wagmi";
+
+/**Chainlink Price Feed. Retrieves OHMETH and ETH/{RESERVE} feed **/
+export const OHMPriceHistory = (assetPair = "OHMv2/ETH") => {
+  const graphURL = "https://api.thegraph.com/subgraphs/name/openpredict/chainlink-prices-subgraph";
+  const {
+    data = [],
+    isFetched,
+    isLoading,
+  } = useQuery(["getOHMPriceHistory", assetPair], async () => {
+    const data = await request(
+      graphURL,
+      gql`
+        {
+          prices(where: { assetPair: "${assetPair}" }, orderBy: timestamp, first: 8, orderDirection: desc) {
+            price
+            timestamp
+          }
+        }
+      `,
+    );
+    return data.prices;
+  });
+
+  return { data, isFetched, isLoading };
+};
+
+export const ReservePriceHistory = (reserveToken: string) => {
+  const graphURL = "https://api.thegraph.com/subgraphs/name/openpredict/chainlink-prices-subgraph";
+  const {
+    data = [],
+    isFetched,
+    isLoading,
+  } = useQuery(["getReservePriceHistory", reserveToken], async () => {
+    const data = await request(
+      graphURL,
+      gql`
+          {
+            prices(where: { assetPair: "${reserveToken}/ETH" }, orderBy: timestamp, first: 8, orderDirection: desc) {
+              price
+              timestamp
+            }
+          }
+        `,
+    );
+    return data.prices;
+  });
+
+  return { data, isFetched, isLoading };
+};
+
+/**
+ * Returns the price of OHM per Reserve Asset
+ * @param reserveToken Reserve Asset
+ * */
+export const PriceHistory = (reserveToken: string) => {
+  const { data: ohmPriceData } = OHMPriceHistory();
+  const { data: reservePriceData } = ReservePriceHistory(reserveToken);
+  const {
+    data = [],
+    isFetched,
+    isLoading,
+  } = useQuery(
+    ["getPriceHistory", ohmPriceData, reservePriceData],
+    () => {
+      const prices = ohmPriceData.map((ohmPrice: { price: number; timestamp: number }, index: any) => {
+        return {
+          price: ohmPrice.price / 1e18 / (reservePriceData[index].price / 1e18),
+          timestamp: new Date(ohmPrice.timestamp * 1000).toLocaleString(),
+        };
+      });
+      return prices;
+    },
+    { enabled: ohmPriceData.length > 0 && reservePriceData.length > 0 },
+  );
+  return { data, isFetched, isLoading };
+};
+
+/**
+ * Returns the current price of the Operator at the given address
+ */
+export const OperatorPrice = () => {
+  const { chain = { id: 1 } } = useNetwork();
+
+  const contract = RANGE_PRICE_CONTRACT.getEthersContract(chain.id);
+  const { data, isFetched, isLoading } = useQuery(["getOperatorPrice", chain], async () => {
+    return parseBigNumber(await contract.getCurrentPrice(), 18);
```

Comment:
must be reverting here; rn getCurrentPrice is 0... not sure if that will ever happen in prod
<img width="198" alt="image" src="https://user-images.githubusercontent.com/80423742/190145170-6a333890-f3f1-4b41-996c-09848361e85b.png">

Reply:
yeah, i have the interface set to 'loading' if we dont return a price.. since everything depends on that. Oighty has a price feed script he runs on goerli to move the price around. In prod this would just be the returned price from the chainlink feed.